supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystemAsyncJobExecutionActor.scala

```diff
@@ -0,0 +1,299 @@
+  }
+
+  def sharedReceive(jobOption: Option[JobType]): Receive = LoggingReceive {
+    case AbortJobCommand =>
+      jobOption foreach tryKill
+  }
+
+  def instantiatedScript: String = {
+    val pathTransformFunction: WdlValue => WdlValue = if (runsOnDocker) toDockerPath else toPath
+    val tryCommand = sharedFileSystem.localizeInputs(jobPaths.callRoot,
+      runsOnDocker, fileSystems, jobDescriptor.inputs) flatMap { localizedInputs =>
+      call.task.instantiateCommand(localizedInputs, callEngineFunction, pathTransformFunction)
+    }
+    tryCommand.get
+  }
+
+  override def executeOrRecover(mode: ExecutionMode)(implicit ec: ExecutionContext) = {
+    Future.fromTry(Try {
+      tellMetadata(startMetadataEvents)
+      executeScript()
+    })
+  }
+
+  private lazy val metadataJobKey = {
+    val jobDescriptorKey: BackendJobDescriptorKey = jobDescriptor.key
+    MetadataJobKey(jobDescriptorKey.call.fullyQualifiedName, jobDescriptorKey.index, jobDescriptorKey.attempt)
+  }
+
+  private def metadataKey(key: String) = MetadataKey(workflowId, Option(metadataJobKey), key)
+
+  private def metadataEvent(key: String, value: Any) = MetadataEvent(metadataKey(key), MetadataValue(value))
+
+  val runtimeAttributesEvents = validatedRuntimeAttributes.attributes map {
+    case (key, value) =>
+      metadataEvent(s"runtimeAttributes:$key", value)
+  }
+
+  def startMetadataEvents: Iterable[MetadataEvent] = runtimeAttributesEvents ++ List(
+    metadataEvent(CallMetadataKeys.Stdout, jobPaths.stdout.toAbsolutePath),
+    metadataEvent(CallMetadataKeys.Stderr, jobPaths.stderr.toAbsolutePath),
+    // TODO: PBE: The REST endpoint toggles this value... how/where? Meanwhile, we read it decide to use the cache...
+    metadataEvent("cache:allowResultReuse", true),
+    metadataEvent(CallMetadataKeys.CallRoot, jobPaths.callRoot)
+  )
+
+  /**
+    * Fire and forget info to the metadata service
+    */
+  def tellMetadata(events: Iterable[MetadataEvent]): Unit = {
+    serviceRegistryActor ! PutMetadataAction(events)
+  }
+
+  def pathPlusSuffix(path: Path, suffix: String) = path.resolveSibling(s"${path.name}.$suffix")
+
+  def stdoutSubmit: PathWriter = pathPlusSuffix(jobPaths.stdout, "submit").untailed
+
+  def stderrSubmit: PathWriter = pathPlusSuffix(jobPaths.stderr, "submit").untailed
+
+  def executeScript(): SharedFileSystemPendingExecutionHandle = {
+    val script = instantiatedScript
+    jobLogger.info(s"`$script`")
+    jobPaths.callRoot.createDirectories()
+    writeScript(script, scriptDir.toAbsolutePath)
+    jobLogger.info(s"command: $processArgs")
+    val stdoutWriter = stdoutSubmit
+    val stderrWriter = stderrSubmit
+    val argv = processArgs.argv.map(_.toString)
+    val proc = argv.run(ProcessLogger(stdoutWriter.writeWithNewline, stderrWriter.writeWithNewline))
+    val run = getJob(proc, stdoutWriter, stderrWriter)
+    // TODO: If returning a future is it ok to change the context.become in a future?
```

Comment:
I'm a giant -1 on that idea. IMO changing the receive method inside a Future is just as bad, if not worse than dealing with an Actor's mutable state from inside a Future

Also in general if people are screwing around w/ become, I'd rather we use the FSM DSL since that's our MO